app/validators/placeholders/placeholder_validator.py

```diff
@@ -144,19 +144,23 @@ def validate_answer_and_transform_unit_match(self, *, arguments, transform_type)
         answer_id = value.get("identifier")
```

Comment:
in the case of the value source being a calculated summary, `value.get("identifier")` is the block_id for the calculated summary, so I think the check should be here and existing code calling it `answer_id` should change

if you reuse the `get_calculation_block_ids` from questionnaire schema you could do something like this and then the new method wouldn't be needed
```
if value.get("source") == "calculated_summary":
    if answer_ids := self.questionnaire_schema.get_calculation_block_ids(
        block=self.questionnaire_schema.get_block(value.get("identifier")),
        source_type="answers",
    ):
        answer_id = answer_ids[0]
```

then existing answers_unit = ... would work

Reply:
agree with passing it into the error at the end 👍 maybe it could be renamed from `answer_id` to `identifier` though? in your new test the `answer_id` is `unit-total-playback` which is the block id for the calculated summary